addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1347,19 +1349,32 @@ registry.WebsiteAnimate = publicWidget.Widget.extend({
                 visible = actualScroll >=
                     totalScrollHeight - heightFromFooter - elHeight + elOffset;
             } else {
-                const scrollTop = document.body.classList.contains("modal-open") ?
-                    this.$target.find('.modal:visible').scrollTop() :
-                    this.$scrollingElement.scrollTop();
-                const elTop = this._getElementOffsetTop($el[0]) - scrollTop;
                 visible = this.windowsHeight > (elTop + elOffset) &&
                     0 < (elTop + elHeight - elOffset);
             }
-            if (visible && state === "paused") {
-                $el.addClass("o_visible");
-                this._startAnimation($el);
-            } else if (!visible && $el.hasClass("o_animate_both_scroll") && state === "running") {
-                $el.removeClass("o_visible");
-                this._resetAnimation($el);
+            if (animateOnScroll) {
+                if (visible) {
+                    const start = 100 / (parseFloat(el.dataset.scrollZoneStart) || 1);
+                    const end = 100 / (parseFloat(el.dataset.scrollZoneEnd) || 1);
+                    const out = el.classList.contains('o_animate_out');
+                    const ratio = (out ? elTop + elHeight : elTop) / (this.windowsHeight - (this.windowsHeight / start));
+                    const duration = parseFloat(window.getComputedStyle(el).animationDuration);
+                    const delay = (ratio - 1) * (duration * end);
+                    el.style.animationDelay = (out ? - duration - delay : delay) + "s";
+                    el.classList.add('o_animating');
+                    this._toggleOverflowXYHidden(true);
+                } else if (el.classList.contains('o_animating')) {
+                    el.classList.remove('o_animating');
+                    this._toggleOverflowXYHidden(false);
```

Comment:
I made it like that for `_toggleOverflowXYHidden` because I dont wnat to call it at each scroll movement when the animate element is not visible.. And so, I use also the else for `o_animating` class, it seems more readable for me like this